consensus/src/consensus/mod.rs

```diff
@@ -447,6 +451,59 @@ impl ConsensusApi for Consensus {
         Ok(self.services.dag_traversal_manager.calculate_chain_path(hash, self.get_sink()))
     }
 
+    /// Returns a Vec of header samples since genesis
+    /// Ordered ascending by daa_score, first entry is genesis
+    fn get_chain_block_samples(&self) -> Vec<DaaScoreTimestamp> {
+        // Sorted from genesis to latest pruning_point_headers
+        let mut sample_headers = Vec::<DaaScoreTimestamp>::new();
```

Comment:
preallocate vector with capacity

Reply:
Please elaborate why

Reply:
because we know that the length should be gte pp_headers + mainnet predefined scores, and we able to avoid vector reallocating

Reply:
Thanks. Handled in [96dced0](https://github.com/kaspanet/rusty-kaspa/pull/268/commits/96dced071bf66843789e8893c89304de9e109b13)